packages/grafana-data/src/types/fieldOverrides.ts

```diff
@@ -48,6 +48,12 @@ export interface FieldConfigEditorConfig<TOptions, TSettings = any, TValue = any
    * @param field
    */
   shouldApply?: (field: Field) => boolean;
+
+  /** Indicates that option shoukd not be available in the Field config tab */
+  hideFromFieldOptions?: boolean;
```

Comment:
Not a fan of this name, happy for any suggestion :)

Reply:
`customizable?: boolean` ?